### PR TITLE
Extract sbt-scripted-tools & dedup scripted tests

### LINKED ISSUE #8588: Unify and publish all common tools for play/lagom scripted tests
Both Play and Lagom scripted tests have custom tasks created to help assert the code reloads, make an HTTP/S request and check the response ([here](https://github.com/lagom/lagom/blob/f5ed4f1f2a0b05f6948489275793573238dab72a/dev/sbt-scripted-tools/src/main/scala/com/lightbend/lagom/sbt/scripted/ScriptedTools.scala#L28) and [here](https://github.com/playframework/playframework/blob/master/framework/src/sbt-plugin/src/sbt-test/play-sbt-plugin/generated-keystore/project/Build.scala)), validate file contents and more...

Lagom already groups and reuses some of them in `lagom-sbt-scripted-tools`.

### PR DESCRIPTION
**Pull request by dwijnand**
Fixes #8588
Builds on #9575
Builds on #9576
Builds on #9577

### CODE REVIEW

File: dev-mode/sbt-plugin/src/sbt-test/evolutions/multiple-databases/build.sbt [hunk 1]
```diff
@@ -10,22 +10,19 @@ lazy val root = (project in file("."))
...
       val path :: Nil = args
-      DevModeBuild.applyEvolutions(path)
+      ScriptedTools.applyEvolutions(path)
     },
     InputKey[Unit]("verifyResourceContains") := {
       val args                         = Def.spaceDelimited("<path> <status> <words> ...").parsed
       val path :: status :: assertions = args
-      DevModeBuild.verifyResourceContains(path, status.toInt, assertions, 0)
+      ScriptedTools.verifyResourceContains(path, status.toInt, assertions)
     }
```
Comment (ignasi35) [inline dev-mode/sbt-plugin/src/sbt-test/evolutions/multiple-databases/build.sbt]: Is there any way to also extract these `Keyp` definitions to a common location?

Comment (dwijnand) [reply dev-mode/sbt-plugin/src/sbt-test/evolutions/multiple-databases/build.sbt]: Yes, further deduplication can be done.  See `ScriptedTools.dumpRoutesSourceOnCompilationFailure` for an example of doing just that.

I self-timeboxed this effort, as it took a long time to merge the 9 distinct versions of `DevModeBuild` and firefight the CI.

With this foundation follow-up PRs with deduplication are very welcome!

File: dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/play-position-mapper/build.sbt [hunk 2]
```diff
@@ -2,24 +2,32 @@
...
-      InputTask.separate[String, Boolean](simpleParser _)(state(s => checkLogContains)).evaluated
-    },
-    TaskKey[Unit]("compileIgnoreErrors") := state.map { state =>
-      Project.runTask(compile in Compile, state)
-    }.value
+      import sbt.complete.DefaultParsers._
+      InputTask.separate[String, Boolean]((_: State) => Space ~> any.+.map(_.mkString(""))) {
+        state(_ => (msg: String) => task {
+          if (ScriptedTools.bufferLoggerMessages.forall(!_.contains(msg))) {
+            sys.error(
```
Comment (marcospereira) [inline dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/play-position-mapper/build.sbt]: We can then delete ` dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/play-position-mapper/project/*.scala` files.

Comment (marcospereira) [reply dev-mode/sbt-plugin/src/sbt-test/play-sbt-plugin/play-position-mapper/build.sbt]: Forget it. I see they were moved.

File: dev-mode/sbt-plugin/src/sbt-test/evolutions/multiple-databases/build.sbt (hunk 1 above)
Comment (ignasi35) [reply dev-mode/sbt-plugin/src/sbt-test/evolutions/multiple-databases/build.sbt]: Makes total sense. Thanks @dwijnand

**Pull request merged**